doc/sonic-application-extension/tpcm_app_ext.md

```diff
@@ -0,0 +1,467 @@
+
+
+### Definitions/Abbreviations 
+
+#### Table 1 Abbreviations
+
+| **Abbreviation** | **Definition**                        |
+| ---------------- | ------------------------------------- |
+| SONiC            | Software for Open Networking in Cloud |
+| DB               | Database                              |
+| API              | Application Programming Interface     |
+| SAI              | Switch Abstraction Interface          |
+| YANG             | Yet Another Next Generation           |
+| JSON             | Java Script Object Notation           |
+| XML              | eXtensible Markup Language            |
+| gNMI             | gRPC Network Management Interface     |
+| TPC              | Third Party Container                 |
+| TPCM             | Third Party Container Management      |
+| AEF              | Application Extension Framework       |
+
+### Document/References
+
+#### Table 2 References
+
+| **Document**                       | **Location**  |
+|------------------------------------|---------------|
+| SONiC Application Extension Infrastructure HLD | [https://github.com/stepanblyschak/SONiC/blob/8d1811f7592812bb5a2cd0b7af5023f5b6449219/doc/sonic-application-extention/sonic-application-extention-hld.md) |
+
+
+### Overview 
+
+SONiC is an open and extensible operating system. SONiC can be extended with Third Party Containers to help with Orchestration, Monitoring, and extending any features or capabilities of SONiC NOS. These can be custom built or readily available in docker community. The rich open eco system provides with multiple docker based applications which can be used in SONiC. And these need not be pre-defined packages in SONiC. While SONiC Application Extensions framework provides the infrastructure to package and integrate a TPC into SONiC, we need more capabilities to provide a dynamic environment and capabilities:
+
+- Dynamically install TPCs from various sources without pre-packaging them in SONIC. 
+- Ensure TPC has right system resources and privileges
+- Abilitiy to configure startup command arguments to TPC
+- Configure resource limits to TPCs, to ensure it does not starve core SONiC containers
+- Ability to seamlessly integrate into SONiC Services Architecture enabling start/ auto restart/ establish dependencies
+
+
+### Requirements
+
+These open TPCs help extend SONiC capabilities, and thus the following requirements are outlined to integrate them seamlessly into SONiC
```

Comment:
As per discussion online, could you please summarize what are the benefits of preparing docker image on the box instead of preparing image outside the box?

Reply:
@sg893052 Thanks, I see one advantage is that the user does not need to have a registry to place his re-built standard docker images and can pull them from a well known existing docker registries.